p9/messages.go

```diff
@@ -1999,6 +2000,177 @@ func (r *rusymlink) String() string {
 	return fmt.Sprintf("Rusymlink{%v}", &r.rsymlink)
 }
 
+// lock - acquire or release a POSIX record lock
+// from the linux kernel: req = p9_client_rpc(clnt, P9_TLOCK, "dbdqqds", fid->fid, flock->type, (etc.)
+// byte doubleword quad quad doubleword string
+// This confirms this doc:
+// size[4] Tlock tag[2] fid[4] type[1] flags[4] start[8] length[8] proc_id[4] client_id[s]
+// start, length, and proc_id correspond to the analogous fields passed to Linux fcntl(F_SETLK):
+// size[4] Rlock tag[2] status[1]
+// lock is used to acquire or release a POSIX record lock on fid and has semantics similar to Linux fcntl(F_SETLK).
+// Most servers implement almost none of this stuff -- they use berkeley flock -- i.e., whole file locking.
+// Hence most of these parameters are ignored. A good thing, since this kind of locking over a network
+// makes no sense.
+
+// LockType is lock type for Tlock
+type LockType uint8
+
+// These constants define Lock operations: Read, Write, and Un(lock)
+const (
+	ReadLock  LockType = iota
+	WriteLock LockType = 1
+	Unlock    LockType = 2
+)
+
+func (l *LockType) String() string {
+	switch *l {
+	case ReadLock:
+		return "ReadLock"
+	case WriteLock:
+		return "WriteLock"
+	case Unlock:
+		return "Unlock"
+	}
+	return "Unknown lock type"
+}
+
+// A few notes on 9p2000.L and locks.
+// Docs are incomplete and confuse whence and flags. Turns out only flags matter.
+// And whence is not in the packet. And the implementations don't use any of this stuff.
+// They just use flock.
+//
+// Locking over a network almost never works anyway. And note that this packet includes a PID.
+// On a system with, say, 1M cores (these exist) a PID is hardly disambiguating.
+// So, this can't work and is hardly worth doing.
+// "something not worth doing is not worth doing well."
+//
+// This is in some docs,
+// struct flock {
+//     short l_type;  /* Type of lock: F_RDLCK, F_WRLCK, F_UNLCK */
+//     short l_whence;/* How to interpret l_start: SEEK_SET, SEEK_CUR, SEEK_END */
+//     off_t l_start; /* Starting offset for lock */
+//     off_t l_len;   /* Number of bytes to lock */
+//     pid_t l_pid;   /* PID of process blocking our lock (F_GETLK only) */
+// };
+// flags bits are:
+
+// #define P9_LOCK_FLAGS_BLOCK 1    /* blocking request */
+// #define P9_LOCK_FLAGS_RECLAIM 2  /* reserved for future use */
+// client_id is an additional mechanism for uniquely identifying the lock requester and is set to the nodename by the Linux v9fs client.
+
+// LockStat contains lock status result.
+type LockStat uint8
```

Comment:
Can you call this LockStatus? I thought it was a statistic or something at first.